chapters/04-function-development.Rmd

```diff
@@ -8,35 +8,115 @@
 - Mostly *how workflow actually looks like*.
 - Building functions up slowly, making small targeted functions that build up
   into a bigger more complex functions
-- Process control (if-else, stop, return, switch)
+- Process control (if, stop)
 - Dependency management
+
+TODO: should this be here or later?
+
 - Function documentation (with roxygen2), part 1
 - To use `@examples` to help with creating function
 
-## Final exercise
+## Learning Objectives
+
+* Convert a chunk of R code into a function
+* Include a function in an R package
+* Provide basic function documentation using Roxygen
+* Use functions from other packages in a function in your own package
+* Return an error message from a function
```

Comment:
Talk about explicit and implicit returns? That you can use either in R but recommend one